internal/charmstore/stats.go

```diff
@@ -574,8 +575,8 @@ func (s *Store) IncrementDownloadCountsAsync(id *charm.Reference) {
 
 // IncrementDownloadCounts updates the download statistics for entity id in both
 // the statistics database and the search database.
-func (s *Store) IncrementDownloadCounts(id *charm.Reference) error {
-	key := EntityStatsKey(id, params.StatsArchiveDownload)
+func (s *Store) IncrementDownloadCounts(id *router.ResolvedURL) error {
+	key := EntityStatsKey(&id.URL, params.StatsArchiveDownload)
```

Comment:
So this updates the stats for a reference with a user defined, but not for a promulgated. Is aggregation on all revisions for "/trusty/mysql" work?